ts/components/session/settings/SessionSettings.tsx

```diff
@@ -622,7 +626,11 @@ export class SettingsView extends React.Component<SettingsViewProps, State> {
         },
         comparisonValue: undefined,
         setFn: async () => {
-          await currentModel.unblock();
+          if (currentModel) {
+            await currentModel.unblock();
+          } else {
+            await BlockedNumberController.unblock(blockedNumber);
+          }
```

Comment:
The issue with this is that the UI won't refresh with an unblock from 
```
            await BlockedNumberController.unblock
```

We need to find a way to have it refreshed

Reply:
Done in the next Commit using "Component.forceUpdate"